### PR TITLE
feat(sbb-journey-header): component refactoring

### LINKED ISSUE #1240: story(sbb-journey-header): review/refactor
### Feature Description
We want to review/refactor the sbb-journey-header implementation.

[Figma Spec](https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?node-id=2153%3A575261)

**Design Spec**

- 

**Technical Spec**

- Remove appearance property
- Rename `isRoundTrip` to `roundTrip`
- Change size property values to `m` and `l`
- Change markup property to `level: '1' | '2' | '3' | '4' | '5' | '6' = '3'`
  - `1`-`6` renders the content in a sbb-title with the appropriate level
- Refactor/remove classes on host
- Implement correct i18n handling

### Definition of Done
 - [ ]  Component is implemented
 - [ ]  Documentation is provided
 - [ ]  Tests are implemented
 - [ ]  E2E Tests are implemented
 - [ ]  Storybook stories are implemented
 - [ ]  Navigation via keyboard is tested
 - [ ]  Screen reader output is tested
 - [ ]  High-contrast is tested
 - [ ]  Remaining accessibility is tested
 - [ ]  UX approved

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #1240 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- component refactoring

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [x] JAWS Firefox Desktop
- [x] JAWS Chrome Desktop
- [x] NVDA Firefox Desktop
- [x] NVDA Chrome Desktop
- [x] VoiceOver Safari Desktop
- [x] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [x] Other (please describe): component refactoring

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

- `appearance` prop has been removed;
- `isRoundTrip` prop has been renamed to `roundTrip`;
- `markup` prop has been renamed to `level` and values has been changed;
- `negative` prop has been added;
- `size` has now `m` and `l` as values instead of `5` and `4`.

### CODE REVIEW

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Thanks for the refactoring! Some open issues.

- home logged in story has to be migrated (usage of journey header 2x)
- sbb-card-product.stories.js has to be migrated (usage of journey header)

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 1]
```diff
@@ -5,58 +5,54 @@
...
   --sbb-title-text-color-normal: var(--sbb-color-charcoal-default);
+  --sbb-journey-header__icon-spacing-inset: var(--sbb-spacing-fixed-1x);
 }
 
-.journey-header[id] {
+:host([negative]:not([negative='false'])) {
+  @include sbb.title--negative;
+}
+
+.sbb-journey-header[id] {
```
[the remark this answers is not in the document]
Comment (DavideMininni-Fincons) [reply src/components/sbb-journey-header/sbb-journey-header.scss]: It works, so I changed it as suggested 🚀

**Review by kyubisation — APPROVED** [review]
LGTM 👍

File: src/components/sbb-journey-header/sbb-journey-header.custom.d.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
 export interface InterfaceJourneyHeaderAttributes {
-  appearance: 'primary' | 'primary-negative';
-  markup: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
-  size: '4' | '5';
+  level: '1' | '2' | '3' | '4' | '5' | '6';
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.custom.d.ts:2]: please remove here and in tsx file import directly from title `InterfaceTitleAttributes['level']`
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 3]
```diff
@@ -5,58 +5,45 @@
 @include sbb.host-component-properties;
 
 :host {
-  --_icon-spacing-inset: var(--sbb-spacing-fixed-1x);
   --sbb-title-text-color-normal: var(--sbb-color-charcoal-default);
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:8]: It's handled in sbb-title.
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 4]
```diff
@@ -5,58 +5,45 @@
...
+  --sbb-journey-header-gap: var(--sbb-spacing-fixed-1x);
 }
 
-.journey-header[id] {
+:host([negative]:not([negative='false'])) {
+  @include sbb.title--negative;
+}
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:14]: It's defined in sbb-title
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 5]
```diff
@@ -5,58 +5,45 @@
...
+
+:host([id]) {
   @include sbb.scroll-margin-block-start;
 }
 
-.journey-header {
+sbb-title {
+  --sbb-title-margin-block-start: 0;
+  --sbb-title-margin-block-end: 0;
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:22]: It's easier like that and also consistent with other cases in code base.
```suggestion
  // Overwrite sbb-title default margin
  margin: 0;
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 6]
```diff
@@ -5,58 +5,45 @@
...
+}
+
+.sbb-journey-header {
   @include sbb.font-smoothing;
+  @include sbb.title-5($exclude-spacing: true);
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:27]:
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 7]
```diff
@@ -5,58 +5,45 @@
...
-  justify-content: space-between;
   align-items: center;
+  gap: var(--sbb-journey-header-gap);
   width: fit-content;
   color: var(--sbb-title-text-color-normal);
 
-  // Right to left expression
+  :host([size='l']) & {
+    @include sbb.title-4($exclude-spacing: true);
+  }
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:38]:
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.scss [hunk 8]
```diff
@@ -5,58 +5,45 @@
...
   @include sbb.font-smoothing;
+  @include sbb.title-5($exclude-spacing: true);
 
   display: flex;
   flex-wrap: wrap;
-  justify-content: space-between;
   align-items: center;
+  gap: var(--sbb-journey-header-gap);
   width: fit-content;
   color: var(--sbb-title-text-color-normal);
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.scss:34]:
```suggestion
```

File: src/components/sbb-journey-header/sbb-journey-header.tsx [hunk 9]
```diff
@@ -10,75 +10,60 @@ import { InterfaceJourneyHeaderAttributes } from './sbb-journey-header.custom';
...
-
-  /** Journey header appearance */
-  @Prop() public appearance?: InterfaceJourneyHeaderAttributes['appearance'] = 'primary';
-
-  /** Journey header size */
-  @Prop() public size?: InterfaceJourneyHeaderAttributes['size'] = '5';
-
-  public render(): JSX.Element {
-    const journeyIcon = this.isRoundTrip ? iconRoundTrip : iconOneWay;
+  @Prop() public level?: InterfaceJourneyHeaderAttributes['level'] = '3';
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.tsx:26]:
```suggestion
  @Prop() public level?: InterfaceTitleAttributes['level'] = '3';
```

File: src/components/sbb-journey-header/sbb-journey-header.tsx [hunk 10]
```diff
@@ -10,75 +10,60 @@ import { InterfaceJourneyHeaderAttributes } from './sbb-journey-header.custom';
...
-  @Prop() public isRoundTrip?: boolean;
+  /** Whether the journey is a round trip. If so, the icon changes to a round-trip one. */
+  @Prop() public roundTrip?: boolean;
 
   /**
-   * Journey header markup:
-   * Depends on the context where the journey will be used but it is important
-   * to pick the correct markup element to match to correct semantics
+   * Journey header markup: depending on the context where it will be used,
+   * it is important to pick the correct markup element to match the correct semantics.
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.tsx:24]:
```suggestion
   * Heading level of the journey header element (e.g. h1-h6).
```

File: src/components/sbb-journey-header/sbb-journey-header.tsx [hunk 11]
```diff
@@ -10,75 +10,60 @@ import { InterfaceJourneyHeaderAttributes } from './sbb-journey-header.custom';
...
     return (
-      <TAGNAME {...attrs} dir={currentWritingMode}>
-        <span class="connection-text-origin connection--visually-hidden">
-          {connectionTextOrigin}
-        </span>
-        <span class="origin">{this.origin}</span>
-        <span class="icon" innerHTML={journeyIcon}>
-          <span class="connection-text-destination connection--visually-hidden">
-            {connectionTextDestination}
+      <sbb-title level={this.level} negative={this.negative}>
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.tsx:45]: You need to transform the size property into the visualLevel property of the sbb-title. I don't know the actual sizes but something like 
```suggestion
      <sbb-title level={this.level} negative={this.negative} visual-level={this.size === 'm' ? '5' : '3'}>
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]
Almost done ;)

Some comments regarding the sbb-title refactoring and the chromatic stories fiel is missing. Please add it.

File: src/components/sbb-journey-header/sbb-journey-header.tsx [hunk 12]
```diff
@@ -42,7 +40,7 @@ export class SbbJourneyHeader {
     const iconName = this.roundTrip ? 'arrows-left-right-small' : 'arrow-long-right-small';
 
     return (
-      <sbb-title level={this.level} negative={this.negative}>
+      <sbb-title level={this.level} negative={this.negative} visual-level={this.size === 'l' ? '3' : '5'}>
```
Comment (jeripeierSBB) [inline src/components/sbb-journey-header/sbb-journey-header.tsx:43]: Is it not level 4?
```suggestion
      <sbb-title level={this.level} negative={this.negative} visual-level={this.size === 'l' ? '3' : '4'}>
```

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thanks

**Pull request merged**